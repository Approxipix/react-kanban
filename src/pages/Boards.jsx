import React, { Component } from 'react'
import { connect } from 'react-redux';
import styled from 'styled-components'
import { NavLink } from 'react-router-dom';
import BoardAdder from '../components/Board/BoardAdder';

const Wrapper = styled.div`
  padding: 1.5rem;
`;

const Title = styled.h1`
  margin-bottom: 1rem;
  font-size: 1.5rem;
  font-weight: 600;
`;

const BoardList = styled.ul`
  display: grid;
  grid-template-columns: repeat(4, minmax(13rem, 1fr));
  grid-auto-rows: 1fr;
  grid-gap: 1rem;
`;

const BoardItem = styled.li`
  padding: .6rem;
  background-color: ${props => props.color};
  border-radius: .4rem
  &:hover {
    box-shadow: inset 0px 0px 0px 100px rgba(0, 0, 0, .1);
  }
`;

const BoardTitle = styled.h2`
  margin-bottom: .5rem;
  font-size: 1rem;
  color: white;
  font-weight: 700;
  text-decoration: none;
  overflow-wrap: break-word;
`;

const BoardScheme = styled.div`
  display: flex;
  height: 6.25rem;
  margin-right: .3rem;
  overflow: hidden;
`;

const BoardSchemeList = styled.div`
  display: inline-block;
  flex-shrink: 0;
  height: ${props => props.height}px;
  width: 1.5rem;
  margin-right: 6px;
  border-radius: 3px;
  background: rgba(255, 255, 255, 0.25);
`;

class Boards extends Component {
  render() {
    const { boards } = this.props;
    return (
      <Wrapper>
        <Title>
          My Boards
        </Title>
        <BoardList>
          {boards.map((board, index) => (
            <NavLink key={index} to={`/board/${index}`}>
              <BoardItem color={board.color}>
                <BoardTitle>
                  {board.title}
                </BoardTitle>
                <BoardScheme>
                  {board.lists.map((list, index) => (
                    <BoardSchemeList
                      key={index}
                      height={Math.min((list.cards.length + 1) * 10, 100)}
                    />
                  ))}
                </BoardScheme>
              </BoardItem>
            </NavLink>
          ))}
          <BoardAdder />
        </BoardList>
      </Wrapper>
    )
  }
}

function mapStateToProps(state) {
  return {
    boards: state.rootReducer.boards,
  }
}

export default connect(mapStateToProps)(Boards);
