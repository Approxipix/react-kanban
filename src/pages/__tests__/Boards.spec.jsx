import React from 'react';
import { shallow } from 'enzyme';
import configureStore from "redux-mock-store";
import toJson from "enzyme-to-json";
import Boards from '../Boards';

const mockStore = configureStore([]);

describe('<Boards>', () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore({
      boardReducer: {
        Board1ID: {
          _boardId: 'Board1ID',
          title: 'Board1 Title',
          color: '#af2232',
          lists: ['List1ID', 'List2ID', 'List3ID'],
        },
        Board2ID: {
          _boardId: 'Board2ID',
          title: 'Board2 Title',
          color: '#af2232',
          lists: ['List4ID'],
        }
      },
      listReducer: {
        List1ID: {
          _listId: 'List1ID',
          title: 'List1 Title',
          cards: []
        },
        List2ID: {
          _listId: 'List2ID',
          title: 'List2 Title',
          cards: []
        },
        List3ID: {
          _listId: 'List3ID',
          title: 'List3 Title',
          cards: []
        },
        List4ID: {
          _listId: 'List4ID',
          title: 'List4 Title',
          cards: []
        }
      },
    });
    const props = {
      cardId: 'CardID',
      checkListId: 'CheckListID',
    };

    const wrapper = shallow(<Boards store={store} {...props} />);
    component = wrapper.find('Boards').dive();
  });

  it('should render with given state from Redux store and correct props', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render a BoardItem for each board in boardReducer', () => {
    expect(component.find('BoardItem').length).toEqual(2)
  });

  it('should render a BoardSchemeBar for each list in board', () => {
    expect(component.find('BoardItem').first().find('BoardSchemeBar').length).toEqual(3)
  });
});
