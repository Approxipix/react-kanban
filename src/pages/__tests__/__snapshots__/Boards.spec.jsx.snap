// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Boards> should render with given state from Redux store and correct props 1`] = `
<styled.div>
  <styled.h1>
    My Boards
  </styled.h1>
  <styled.ul>
    <NavLink
      key="0"
      to="/b/Board1ID"
    >
      <BoardItem
        color="#af2232"
      >
        <styled.h2>
          Board1 Title
        </styled.h2>
        <styled.div>
          <BoardSchemeBar
            height={10}
            key="0"
          />
          <BoardSchemeBar
            height={10}
            key="1"
          />
          <BoardSchemeBar
            height={10}
            key="2"
          />
        </styled.div>
      </BoardItem>
    </NavLink>
    <NavLink
      key="1"
      to="/b/Board2ID"
    >
      <BoardItem
        color="#af2232"
      >
        <styled.h2>
          Board2 Title
        </styled.h2>
        <styled.div>
          <BoardSchemeBar
            height={10}
            key="0"
          />
        </styled.div>
      </BoardItem>
    </NavLink>
    <Connect(BoardAdd) />
  </styled.ul>
</styled.div>
`;
